pymc4/random_variables/continuous.py

```diff
@@ -14,15 +14,516 @@
+
+       f(x \mid \beta) = \frac{2}{\pi \beta [1 + (\frac{x}{\beta})^2]}
+
+    .. plot::
+
+        import matplotlib.pyplot as plt
+        import numpy as np
+        import scipy.stats as st
+        plt.style.use('seaborn-darkgrid')
+        x = np.linspace(0, 5, 200)
+        for b in [0.5, 1.0, 2.0]:
+            pdf = st.cauchy.pdf(x, scale=b)
+            plt.plot(x, pdf, label=r'$\beta$ = {}'.format(b))
+        plt.xlabel('x', fontsize=12)
+        plt.ylabel('f(x)', fontsize=12)
+        plt.legend(loc=1)
+        plt.show()
+
+    ========  ========================
+    Support   :math:`x \in [0, \infty)`
+    Mode      0
+    Mean      undefined
+    Variance  undefined
+    ========  ========================
+
+    Parameters
+    ----------
+    beta : float
+        Scale parameter (beta > 0).
+
+    Developer Notes:
+    ----------------
+    Parameter mappings to TensorFlow Probability are as follows:
+
+    - beta: scale
+
+    In PyMC3, HalfCauchy's location was always zero. However, in a future PR, this can be changed.
+    """
+
+    def _base_dist(self, beta, *args, **kwargs):
+        return tfd.HalfCauchy(loc=0, scale=beta)
+
+
+class HalfNormal(RandomVariable):
+    R"""
+    Half-normal random variable.
+
+    The pdf of this distribution is
+
+    .. math::
+
+       f(x \mid \tau) =
+           \sqrt{\frac{2\tau}{\pi}}
+           \exp\left(\frac{-x^2 \tau}{2}\right)
+       f(x \mid \sigma) =\sigma
+           \sqrt{\frac{2}{\pi}}
+           \exp\left(\frac{-x^2}{2\sigma^2}\right)
+
+    .. note::
+
+       The parameters ``sigma``/``tau`` (:math:`\sigma`/:math:`\tau`) refer to
+       the standard deviation/precision of the unfolded normal distribution, for
+       the standard deviation of the half-normal distribution, see below. For
+       the half-normal, they are just two parameterisation :math:`\sigma^2
+       \equiv \frac{1}{\tau}` of a scale parameter
+
+    .. plot::
+
+        import matplotlib.pyplot as plt
+        import numpy as np
+        import scipy.stats as st
+        plt.style.use('seaborn-darkgrid')
+        x = np.linspace(0, 5, 200)
+        for sigma in [0.4, 1., 2.]:
+            pdf = st.halfnorm.pdf(x, scale=sigma)
+            plt.plot(x, pdf, label=r'$\sigma$ = {}'.format(sigma))
+        plt.xlabel('x', fontsize=12)
+        plt.ylabel('f(x)', fontsize=12)
+        plt.legend(loc=1)
+        plt.show()
+
+    ========  ==========================================
+    Support   :math:`x \in [0, \infty)`
+    Mean      :math:`\sqrt{\dfrac{2}{\tau \pi}}` or :math:`\dfrac{\sigma \sqrt{2}}{\sqrt{\pi}}`
+    Variance  :math:`\dfrac{1}{\tau}\left(1 - \dfrac{2}{\pi}\right)` or :math:`\sigma^2\left(1 - \dfrac{2}{\pi}\right)`
+    ========  ==========================================
+
+    Parameters
+    ----------
+    sigma : float
+        Scale parameter :math:`sigma` (``sigma`` > 0) (only required if ``tau`` is not specified).
+
+    Examples
+    --------
+    .. code-block:: python
```

Comment:
pymc3 code spotted

Reply:
Thanks @ferrine, I'll be sure to check at the end!